src/Messages/DTO/MessagePart.php

```diff
@@ -260,15 +297,26 @@ public function toArray(): array
      */
     public static function fromArray(array $array): self
     {
+        if (isset($array[self::KEY_CHANNEL])) {
+            if (!MessagePartChannelEnum::isValidValue($array[self::KEY_CHANNEL])) {
+                throw new InvalidArgumentException(
+                    sprintf('Invalid channel value: %s', $array[self::KEY_CHANNEL])
+                );
+            }
+            $channel = MessagePartChannelEnum::from($array[self::KEY_CHANNEL]);
```

Comment:
I don't think you need to throw an exception here, as the `from` method will throw an exception. Unless you're wanting the custom exception message?